src/synthesizer/dust/attenuation.py

```diff
@@ -39,26 +39,35 @@ def N09_tau(lam, slope, cent_lam, ampl, gamma):
     """
 
     # Wavelength in microns
-    tmp_lam = np.arange(0.12, 2.2, 0.001)
+    lam_micron = lam.to(um).value
     lam_v = 0.55
-    k_lam = np.zeros_like(tmp_lam)
-
-    ok = (tmp_lam < 0.63)
-    k_lam[ok] = -2.156 + (1.509 / tmp_lam[ok]) \
-        - (0.198 / tmp_lam[ok] ** 2) \
-        + (0.011 / tmp_lam[ok] ** 3)
-    k_lam[~ok] = -1.857 + (1.040 / tmp_lam[~ok])
+    k_lam = np.zeros_like(lam_micron)
+
+    ok1 = (lam_micron >= 0.12) * (lam_micron < 0.63)
+    ok2 = (lam_micron >= 0.63) * (lam_micron < 31.)
+    ok3 = (lam_micron < 0.12)
+    if np.sum(ok1) > 0:
+        k_lam[ok1] = -2.156 + (1.509 / lam_micron[ok1]) \
+            - (0.198 / lam_micron[ok1]**2) \
+            + (0.011 / lam_micron[ok1]**3)
+        func = interpolate.interp1d(lam_micron[ok1], k_lam[ok1],
+                                fill_value="extrapolate")
+    if np.sum(ok2) > 0:
+        k_lam[ok2] = -1.857 + (1.040 / lam_micron[ok2])
+    if np.sum(ok3) > 0:
+        k_lam[ok3] = func(lam_micron[ok3])
+    
     k_lam = 4.05 + 2.659 * k_lam
+    k_v = 4.05 + 2.659 * (-2.156 + (1.509 / lam_v)
```

Comment:
Do these values come from Noll+09 or Calzetti+00 ?

Reply:
These are from Calzetti, the UV bump is the addition from Noll+09